test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs

```diff
@@ -6975,6 +6975,73 @@ public async Task TestPacMan_PreviewInstallPackage_BuildIntegrated_MissingPath_T
             }
         }
 
+        /// <summary>
+        /// Repro for a bug caused by a NullReferenceException being thrown due to a null <see cref="PackageIdentity.Version"/>
```

Comment:
What kind of value does this test provide us beyond being great for this review?

It's an `input` error, not necessarily a contract we're going to try to maintain.

Reply:
The input error is "by design". So you either handle the input error (I did not modify the responsible code)
, or you have a test proving it does what you think it does. 
It's also a test that can be easily changed if/when we decide on a different design.

Reply:
Why did you decide against adding an input check? 
Will the call *always* fail if the input PackageIdentity has a null version?

Reply:
Because I'm trying to fix a bug specifically. Not redesign the inners of Package Manager. 
Not always, nope. Has to be BuildIntegrated project, with at least 1 update available.

Reply:
> Because I'm trying to fix a bug specifically. Not redesign the inners of Package Manager.
 
 I think this can be considered in scope. 
 Changing the NuGetPackageManager class to ensure we don't run into unexpected exceptions is in scope. 
 Especially because there wouldn't be any side-effects.

Reply:
Synced up offline. 

Call it good for 16.9, consider additional quality improvements in NuGetPackageManager for 16.10.